### DISCUSSION TITLE
Make "niri msg action spawn" support quoted substrings

### DISCUSSION THREAD

**Discussion by efeme4** [YaLTeR/niri#607] [Ideas]
This would make niri msg behave more simliarly to i3msg or swaymsg and make it compatible with i3-dmenu-desktop, a wrapper script for dmenu (or wmenu) that uses .desktop files instead of just using the system PATH.

**Comment by YaLTeR** [comment]
What do you mean by support quoted strings?

**Reply by efeme4** [reply]
"This is a string "This is a substring inside quotes" rest of the string" both swaymsg and i3-msg are able to recongnize this and separate elements inside the string and that's why i3-dmenu desktop works on both i3 and sway but not on niri, because niri reads the entire string as one single thing

EDIT: Yeah I just read the issue title, I meant to say SUBstring.

**Reply by YaLTeR** [reply]
niri doesn't use a shell for spawning, which is why this doesn't work

**Comment by efeme4** [comment]
Oh, ok, thanks anyway
